system/coldboxModifications/Controller.cfc

```diff
@@ -133,37 +134,126 @@ component extends="coldbox.system.web.Controller" {
 			return rendered;
 		}
 
-		var result        = "";
-		var view          = "";
-		var handler       = arguments.event;
-		var defaultAction = getSetting( name="EventAction", fwSetting=true, defaultValue="index" );
-		var hndlrExists   = handlerExists( handler );
+		var result          = "";
+		var view            = ListChangeDelims( arguments.event, "/", "." );
+		var deferredViewlet = "";
+		var viewletArgs     = arguments.args;
+		var handler         = arguments.event;
+		var defaultAction   = getSetting( name="EventAction", fwSetting=true, defaultValue="index" );
+		var hndlrExists     = handlerExists( handler );
 
 		if ( !hndlrExists ) {
 			handler = ListAppend( handler, defaultAction, "." );
 			hndlrExists = handlerExists( handler );
 		}
 
 		if ( hndlrExists ) {
-			return runEvent(
+			var requestContext = getRequestContext();
+			requestContext.pushViewletContext( view );
+
+			var result = runEvent(
 				  event          = handler
 				, prepostExempt  = arguments.prepostExempt
 				, private        = arguments.private
-				, eventArguments = { args = arguments.args }
+				, eventArguments = { args = viewletArgs }
 			);
+
+			if ( IsNull( local.result ) ) {
+				view            = requestContext.getViewletView();
+				deferredViewlet = requestContext.getDeferredViewlet();
+				viewletArgs     = requestContext.getViewletArgs( viewletArgs );
+			}
+			requestContext.popViewletContext();
```

Comment:
Can we move this higher up and return early? (ie before if ( IsNull( local.result ) )

if ( !IsNull( local.result ) ) {

Reply:
No, because `requestContext.getViewletView()`, etc. all depend on the viewlet context being the current viewlet. Can/must pop it after that.